akka-actor/src/main/scala/akka/io/dns/DnsProtocol.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns
+
+import akka.io.dns.protocol.{ DnsRecordType, DnsResourceRecord }
+
+import scala.collection.immutable
+
+/**
+ * Supersedes [[akka.io.Dns]] protocol.
+ *
+ * Note that one MUST configure `akka.io.dns.resolver = async` to make use of this protocol and resolver.
+ *
+ * Allows for more detailed lookups, by specifying which records should be checked,
+ * and responses can more information than plain IP addresses (e.g. ports for SRV records).
+ */
+object DnsProtocol {
```

Comment:
The names are to be changed here; I added Dns prefix everywhere while developing, but it likely will be fine without it. Just want to avoid confusion with existing protocol hm. But if we adapt it it would not matter too much